autodiff/function.py

```diff
@@ -17,58 +17,105 @@
 """
```

Comment:
Please can you wirte a general doc that explains that functions transforms variable to variable AND that currently only th following functoins are implemented (exp, sin,...)